support-lambdas/stripe-intent/cfn.yaml

```diff
@@ -0,0 +1,95 @@
+AWSTemplateFormatVersion: '2010-09-09'
+Transform: AWS::Serverless-2016-10-31
+Description: Merges together the salesforce account referenced by a set of zuora accounts
+
+Parameters:
+  Stage:
+    Description: Stage name
+    Type: String
+    AllowedValues:
+      - PROD
+      - CODE
+    Default: CODE
+
+Mappings:
+  StageMap:
+    CODE:
+      DomainName: stripe-intent-code.support.guardianapis.com
+      StripeConfig: test
+    PROD:
+      DomainName: stripe-intent.support.guardianapis.com
```

Comment:
we (@tomrf1 @JoemG ) had a very brief discussion about appropriate host names for prod/code as they are making a similar style lambda. We have the existing code.support.dev-guardianapis.com convention but no certificate so this one is a bit easier.   But now would be a good time for objections.   These conventions are also present in support-service-lambdas repo.